libs/wingsdk/src/shared-aws/bucket.inflight.ts

```diff
@@ -200,15 +201,34 @@ export class BucketClient implements IBucketClient {
     }
     return list;
   }
+  /**
+   * checks if the bucket is public
+   * @returns true if the bucket is public and false otherwise
+   */
+  private async checkIfPublic(): Promise<boolean> {
+    const command = new GetPublicAccessBlockCommand({
+      Bucket: this.bucketName,
+    });
+    const resp: GetPublicAccessBlockCommandOutput = await this.s3Client.send(
+      command
+    );
+    return (
+      !resp.PublicAccessBlockConfiguration?.BlockPublicAcls &&
+      !resp.PublicAccessBlockConfiguration?.BlockPublicPolicy &&
+      !resp.PublicAccessBlockConfiguration?.RestrictPublicBuckets &&
+      !resp.PublicAccessBlockConfiguration?.IgnorePublicAcls
+    );
+  }
 
   /**
    * Returns a url to the given file.
    * @Throws if the file is not public or if object does not exist.
    */
   public async publicUrl(key: string): Promise<string> {
```

Comment:
Do we need to add permissions for this in the IAM policy?

Reply:
yes, I added those, and also fixed the permissions of the other methods (as the tests were failing)